src/main/scala/com/gu/anghammarad/models/Serialization.scala

```diff
@@ -1,38 +1,74 @@
 package com.gu.anghammarad.models
 
+import com.gu.anghammarad.Enrichments._
 import io.circe.Decoder.Result
 import io.circe._
-import io.circe.generic.semiauto._
 import io.circe.parser._
-import io.circe.syntax._
 
 import scala.util.{Failure, Success, Try}
-import com.gu.anghammarad.Enrichments._
 
 
 object Serialization {
-  def parseMapping(jsonStr: String): Try[Mapping] = {
-    val tryJson = parse(jsonStr).fold(
+
+  def parseAllMappings(jsonStr: String): Try[List[Mapping]] = {
+    val allMappings = for {
+      json <- parse(jsonStr)
+      rawMappings <- json.hcursor.downField("mappings").as[List[Json]]
+      mappings <- rawMappings.traverseE(parseMapping)
+    } yield mappings
+
+    allMappings.fold(
       err => Failure(err),
-      json => Success(json)
+      suc => Success(suc)
     )
+  }
 
+  private[models] def parseMapping(json: Json): Result[Mapping] = {
+    val hCursor = json.hcursor
+    for {
+      rawTargets <- hCursor.downField("target").as[Json]
+      rawContacts <- hCursor.downField("contacts").as[Json]
+      targets = parseTargets(rawTargets)
+      contacts = parseContacts(rawContacts)
+    } yield Mapping(targets, contacts)
+  }
 
+  private[models] def parseTargets(jsonTargets: Json): List[Target] = {
+    def parseTarget(key: String, value: String): Option[Target] = {
```

Comment:
Is there a reason why this fn is inlined? It could be a separate function and then it would be independently testable.